File: mobile/src/main/java/treehou/se/habit/ui/colorpicker/ColorpickerActivity.java
package treehou.se.habit.ui.colorpicker;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import com.google.gson.Gson;

import javax.inject.Inject;

import se.treehou.ng.ohcommunicator.connector.models.OHWidget;
import se.treehou.ng.ohcommunicator.util.GsonHelper;
import treehou.se.habit.HabitApplication;
import treehou.se.habit.R;
import treehou.se.habit.util.Settings;

public class ColorpickerActivity extends AppCompatActivity {

    public static final String EXTRA_SERVER = "EXTRA_SERVER";
    public static final String EXTRA_WIDGET = "EXTRA_SITEMAP";
    public static final String EXTRA_COLOR  = "EXTRA_COLOR";

    @Inject Settings settings;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        ((HabitApplication) getApplication()).component().inject(this);
        setTheme(settings.getThemeResourse());
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_colorpicker);

        Bundle bundle = getIntent().getExtras();
        long serverId = bundle.getLong(EXTRA_SERVER);
        String jWidget = bundle.getString(EXTRA_WIDGET);
        int color = bundle.getInt(EXTRA_COLOR);

        Gson gson = GsonHelper.createGsonBuilder();
        OHWidget widget = gson.fromJson(jWidget, OHWidget.class);

        if (savedInstanceState == null) {
            getFragmentManager().beginTransaction()
                    .add(R.id.container, LightFragment.newInstance(serverId, widget, color))
                    .commit();
        }
    }
}

File: mobile/src/main/java/treehou/se/habit/ui/colorpicker/ColorpickerActivity.kt
package treehou.se.habit.ui.colorpicker

import android.os.Bundle

import com.google.gson.Gson

import javax.inject.Inject

import se.treehou.ng.ohcommunicator.connector.models.OHWidget
import se.treehou.ng.ohcommunicator.util.GsonHelper
import treehou.se.habit.BaseActivity
import treehou.se.habit.HabitApplication
import treehou.se.habit.R
import treehou.se.habit.util.Settings

class ColorpickerActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        (application as HabitApplication).component().inject(this)
        setTheme(settings.themeResourse)
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_colorpicker)

        val bundle = intent.extras
        val serverId = bundle!!.getLong(EXTRA_SERVER)
        val jWidget = bundle.getString(EXTRA_WIDGET)
        val color = bundle.getInt(EXTRA_COLOR)

        val gson = GsonHelper.createGsonBuilder()
        val widget = gson.fromJson(jWidget, OHWidget::class.java)

        if (savedInstanceState == null) {
            supportFragmentManager.beginTransaction()
                    .add(R.id.container, LightFragment.newInstance(serverId, widget, color))
                    .commit()
        }
    }

    companion object {

        val EXTRA_SERVER = "EXTRA_SERVER"
        val EXTRA_WIDGET = "EXTRA_SITEMAP"
        val EXTRA_COLOR = "EXTRA_COLOR"
    }
}
